app/src/main/java/org/mozilla/fenix/library/history/viewholders/HistoryDeleteButtonViewHolder.kt

```diff
@@ -6,29 +6,26 @@ package org.mozilla.fenix.library.history.viewholders
 
 import android.view.View
 import androidx.recyclerview.widget.RecyclerView
-import io.reactivex.Observer
 import kotlinx.android.synthetic.main.delete_history_button.view.*
 import org.mozilla.fenix.R
-import org.mozilla.fenix.library.history.HistoryAction
+import org.mozilla.fenix.library.history.HistoryInteractor
 import org.mozilla.fenix.library.history.HistoryState
 
 class HistoryDeleteButtonViewHolder(
     view: View,
-    private val actionEmitter: Observer<HistoryAction>
+    historyInteractor: HistoryInteractor
 ) : RecyclerView.ViewHolder(view) {
     private var mode: HistoryState.Mode? = null
     private val buttonView = view.delete_history_button
 
     init {
         buttonView.setOnClickListener {
             mode?.also {
-                val action = when (it) {
-                    is HistoryState.Mode.Normal -> HistoryAction.Delete.All
-                    is HistoryState.Mode.Editing -> HistoryAction.Delete.Some(it.selectedItems)
+                when (it) {
+                    is HistoryState.Mode.Normal -> historyInteractor.onDeleteAll()
+                    is HistoryState.Mode.Editing -> historyInteractor.onDeleteSome(it.selectedItems)
                     is HistoryState.Mode.Deleting -> null
                 } ?: return@also
```

Comment:
We can get rid of this return and the case for `HistoryState.Mode.Deleting` since we’re not assigning the result to `action` anymore 👍